evm_loader/program/src/account_storage.rs

```diff
@@ -0,0 +1,253 @@
+            clock_account,
+            account_infos: account_infos,
+        })
+    }
+
+    pub fn get_account_by_index(&self, index: usize) -> Option<&SolidityAccount<'a>> {
+        if let Some(acc) = &self.accounts[index] {
+            Some(&acc)
+        } else {
+            None
+        }
+    }
+
+    pub fn get_account_by_index_mut(&mut self, index: usize) -> Option<&SolidityAccount<'a>> {
+        if let Some(acc) = &self.accounts[index] {
+            Some(&acc)
+        } else {
+            None
+        }
+    }
+
+    fn find_account(&self, address: H160) -> Option<usize> {
+        let aliases = self.aliases.borrow();
+        match aliases.binary_search_by_key(&address, |v| v.0) {
+            Ok(pos) => {
+                debug_print!(&("Found account for ".to_owned() + &address.to_string() + " on position " + &pos.to_string()));
+                Some(aliases[pos].1)
+            }
+            Err(_) => {
+                debug_print!(&("Not found account for ".to_owned() + &address.to_string()));
+                None
+            }
+        }
+    }
+
+    fn get_account(&self, address: H160) -> Option<&SolidityAccount<'a>> {
+        if let Some(pos) = self.find_account(address) {
+            self.accounts[pos].as_ref()
+        } else {
+            None
+        }
+    }
+
+    // fn get_account_mut(&mut self, address: H160) -> Option<(&mut SolidityAccount<'a>, usize)> {
+    //     if let Some(pos) = self.find_account(address) {
+    //         self.accounts[pos].as_mut()
+    //     } else {
+    //         None
+    //     }
+    // }
+
+    fn is_solana_address(&self, code_address: &H160) -> bool {
+        *code_address == Self::system_account()
+    }
+
+    fn is_ecrecover_address(&self, code_address: &H160) -> bool {
+        *code_address == Self::system_account_ecrecover()
+    }
+
+    pub fn system_account() -> H160 {
+        H160::from_slice(&[
+            0xffu8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+            0u8, 0u8, 0u8,
+        ])
+    }
+
+    pub fn system_account_ecrecover() -> H160 {
+        H160::from_slice(&[
+            0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+            0u8, 0u8, 0x01u8,
+        ])
+    }
+
+    pub fn apply<A, I>(&mut self, values: A, delete_empty: bool, skip_addr: Option<(H160, bool)>) -> Result<(), ProgramError>
+    where
+        A: IntoIterator<Item = Apply<I>>,
+        I: IntoIterator<Item = (H256, H256)>,
+    {
+        let ether_addr = skip_addr.unwrap_or_else(|| (H160::zero(), true));
+        let system_account = Self::system_account();
+        let system_account_ecrecover = Self::system_account_ecrecover();
+
+        for apply in values {
+            match apply {
+                Apply::Modify {address, basic, code, storage, reset_storage} => {
+                    if (address == system_account) || (address == system_account_ecrecover) {
```

Comment:
`system_account` & `system_account_ecrecover` should be in SolanaBackend only. Here we can call the SolanaBackend static method to check that we do not need to update these accounts.